### PR TITLE
man: add brief documentation for the (sd-pam) processes created due t…

### PR DESCRIPTION
**Pull request by poettering**
…o PAMName=

A follow-up for #4942, adding a brief but more correct explanation of
the processes.

### CODE REVIEW

File: man/systemd.exec.xml [hunk 1]
```diff
@@ -859,14 +859,16 @@
...
-        <listitem><para>Sets the PAM service name to set up a session
-        as. If set, the executed process will be registered as a PAM
-        session under the specified service name. This is only useful
-        in conjunction with the <varname>User=</varname> setting. If
-        not set, no PAM session will be opened for the executed
-        processes. See
-        <citerefentry project='man-pages'><refentrytitle>pam</refentrytitle><manvolnum>8</manvolnum></citerefentry>
-        for details.</para></listitem>
+        <listitem><para>Sets the PAM service name to set up a session as. If set, the executed process will be
+        registered as a PAM session under the specified service name. This is primarily useful in conjunction with the
```
Comment (fsateler) [inline man/systemd.exec.xml]: This changed "only useful with" with "primarily useful with". But, [this option is ignored if no username is specified](https://github.com/systemd/systemd/blob/357cd5563a40a65306901ea2c7cbf121244b76e9/src/core/execute.c#L2706):

```c
if (context->pam_name && username) {
    r = setup_pam(context->pam_name, username, uid, gid, context->tty_path, &accum_env, fds, n_fds);
```

Comment (poettering) [reply man/systemd.exec.xml]: indeed. will fix

File: man/systemd.exec.xml [hunk 2]
```diff
@@ -859,14 +859,17 @@
...
         <term><varname>PAMName=</varname></term>
-        <listitem><para>Sets the PAM service name to set up a session
-        as. If set, the executed process will be registered as a PAM
-        session under the specified service name. This is only useful
-        in conjunction with the <varname>User=</varname> setting. If
-        not set, no PAM session will be opened for the executed
-        processes. See
-        <citerefentry project='man-pages'><refentrytitle>pam</refentrytitle><manvolnum>8</manvolnum></citerefentry>
-        for details.</para></listitem>
+        <listitem><para>Sets the PAM service name to set up a session as. If set, the executed process will be
```
Comment (arvidjaar) [inline man/systemd.exec.xml:862]: > set up a session as.

What `as` means here? This sounds like a typo carried over from previous version?

Comment (poettering) [reply man/systemd.exec.xml:862]: when registering a PAM session you specify a "service name", which PAM then uses to find the PAM configuration file. With PAMName= you configure that service name, i.e. the service identity to use for session registration, and hence we set up a session *as* that service name.  This hence appears correct to me the way it is.

Comment (martinpitt) [reply man/systemd.exec.xml:862]: @arvidjaar : It's just sloppy English with dangling propositions :-) A more legible variant might be "Sets the PAM service name for the session". But this is nitpicking and not a grammar regression, so I'll merge that for now.

**Pull request merged**